### DISCUSSION TITLE
(Issue?) Errors with Avalonia XAML Preview in Visual Studio 2022

### DISCUSSION THREAD

**Discussion by OttoKH** [AvaloniaUI/Avalonia#7598] [Q&A] [answered]
Hello!

Before I explain the issue I'm having, I want to say that I'm still relatively new to cross-platform development, so I might be missing something INCREDIBLY obvious. I'm trying to write a program in C# that accesses various hardware devices on a Raspberry Pi and displays their status on a UI using Avalonia. I'm using Visual Studio 2022 Enterprise as an IDE, then pushing the program to the pi using SSH-Deploy to test. 

What's happening is whenever I start add an initialization command for the wiring library, the XAML preview for my main window's UI breaks, and I get an error in the general output saying:
 **_Unable to load DLL 'libc' or one of its dependencies: The specified module could not be found. (0x8007007E)_**
Weirdly enough, this error doesn't prevent building, and when I push it to the Pi it loads up perfectly, so it only seems to affect my ability to preview the xaml in VS, which is ultimately more of an inconvenience than anything. 

Below is all the code that I need to replicate the issue. 

MainWindow.axaml.cs
```
using Avalonia;
using Avalonia.Controls;
using Avalonia.Interactivity;
using Avalonia.Markup.Xaml;
using Unosquare.RaspberryIO;
using Unosquare.RaspberryIO.Abstractions;
using Unosquare.WiringPi;

namespace AFSD
{
    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();


            // Initialize VLC and Raspberry Pi wiring
            Pi.Init<BootstrapWiringPi>();

        }

        private void InitializeComponent()
        {
            AvaloniaXamlLoader.Load(this);
        }
    }
}

```
MainWindow.axaml
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AFSD.MainWindow"
        Title="AFSD">
	<Panel HorizontalAlignment="Center" VerticalAlignment="Top">
			<Button>
				<TextBlock TextWrapping="Wrap">Foo</TextBlock>
			</Button>
	</Panel>
</Window>
```
If I comment out `Pi.Init<BootstrapWiringPi>();`, then the error goes away and I can view the preview again just fine. I've checked dependencies and nothing seems to be off at all, so I'm at a complete loss as to where to go with this.  

Any ideas?

**Comment by jp2masa** [comment, marked as answer]
You probably need to add a check like this:

```cs
if (!Design.IsDesignMode)
{
    Pi.Init<BootstrapWiringPi>();
}
```

This is because the previewer will create a `MainWindow`, so `Pi.Init` will be called and it doesn't seem to be compatible with the dev machine.
